test/e2e/kube_metrics_adapter_test.go

```diff
@@ -339,7 +339,8 @@ func podHPA(deploymentName string, ingressName string, metricTargets map[string]
 					Kind:       "Ingress",
 					Name:       ingressName,
 				},
-				TargetValue: *resource.NewQuantity(target, resource.DecimalSI),
+				TargetValue:  *resource.NewQuantity(target, resource.DecimalSI),
+				AverageValue: resource.NewQuantity(target, resource.DecimalSI),
```

Comment:
ooc, do you know why the targetValue uses a pointer and averageValue not?

Reply:
Don't ask me why they created such a crazy API. The idea is the `averageValue` is optional. But if you set it you also have to set `targetValue` it just won't take any effect when `averageValue` is set. :man_shrugging: